projects/jdwp/defs/schema.py

```diff
@@ -32,30 +31,63 @@ class PrimitiveType(Enum):
     LOCATION = "location"
 
 
-class Array(Enum):
+class IntegralType(Enum):
+    """Integral type class."""
+
+    INT = "int"
+    BYTE = "byte"
+
+
+@dataclass(frozen=True)
+class ArrayLength:
+    """Array length class."""
+
+    type: IntegralType
+
+
+@dataclass(frozen=True)
+class Array:
     """Array class type."""
 
-    pass
+    element_type: Struct | TaggedUnion
+    length: Field[ArrayLength]
 
 
-class TaggedUnion(Enum):
+@dataclass(frozen=True)
+class TaggedUnion:
     """Tagged Union class type"""
 
-    pass
+    tag: Field[UnionTag]
+    cases: Dict[ModifierKind, Struct]
```

Comment:
We should make `TaggedUnion` generic over `M`/`EnumT` as well. How about something like this:
```
class TaggedUnion[Generic[EnumT]]:
    """Tagged Union class type"""

    tag: Field[UnionTag[EnumT]]
    cases: Mapping[EnumT, Struct]
```

Note that it's better to use `collections.abc.Mapping` than `typing.Dict`